fan-xiaomi.js

```diff
@@ -105,305 +104,324 @@ class FanXiaomi extends HTMLElement {
+        }
+
+        // LED mode event bindings
+        ui.querySelector('.var-led').onclick = () => {
+            this.log('Led')
+            if (ui.querySelector('.fanbox').classList.contains('active')) {
+                let u = ui.querySelector('.var-led')
+                if (u.classList.contains('active') === false) {
+                    this.log(`Set led mode to: On`)
+                    hass.callService(this.config.platform, 'fan_set_led_on', {
+                        entity_id: entityId
+                    });
+                } else {
+                    this.log(`Set led mode to: Off`)
+                    hass.callService(this.config.platform, 'fan_set_led_off', {
                         entity_id: entityId
                     });
                 }
-            } else {
-                ui.querySelector('.var-title').onclick = () => {
-                    this.log('Dialog box')
-                    moreInfo(entityId);
+            }
+        }
+
+        // Oscillation toggle event bindings
+        ui.querySelector('.var-oscillating').onclick = () => {
+            this.log('Oscillate')
+            if (ui.querySelector('.fanbox').classList.contains('active')) {
+                let u = ui.querySelector('.var-oscillating')
+                if (u.classList.contains('active') === false) {
+                    this.log(`Set oscillation to: On`)
+                    hass.callService('fan', 'oscillate', {
+                        entity_id: entityId,
+                        oscillating: true
+                    });
+                } else {
+                    this.log(`Set oscillation to: Off`)
+                    hass.callService('fan', 'oscillate', {
+                        entity_id: entityId,
+                        oscillating: false
+                    });
                 }
             }
-            /*
+        }
+        
+        //Fan title works as on/off button when animation is disabled
+        if (this.config.disable_animation) {
+            ui.querySelector('.var-title').onclick = () => {
+                this.log('Toggle')
+                hass.callService('fan', 'toggle', {
+                    entity_id: entityId
+                });
+            }
+        } else {
             ui.querySelector('.var-title').onclick = () => {
                 this.log('Dialog box')
-                card.querySelector('.dialog').style.display = 'block'
-            }*/
-            this.card = card;
-            this.appendChild(card);
+                moreInfo(entityId);
+            }
+        }
+        /*
+        ui.querySelector('.var-title').onclick = () => {
+            this.log('Dialog box')
+            card.querySelector('.dialog').style.display = 'block'
+        }*/
+        this.card = card;
+        this.appendChild(card);
+    }
+
+    updateUI(hass) {
+        const state = hass.states[this.config.entity];
+
+        if (state.state === 'unavailable') {
```

Comment:
I'm wondering if we should modify this logic or simplify the other check above in Line 125.

```
if (state === undefined || state.state === 'unavailable')
```

Typically I cannot see a scenario where `state` is undefined, the logic was inherited from the pre-fork code. Even the official docs recommend checking `state.state` right away in their examples.

What do you think?

Reply:
Then again I may have answered my own question. 😅  If the entity does not exist, then state would be undefined.